alphabase/psm_reader/psm_reader.py

```diff
@@ -257,33 +265,39 @@ def set_modification_mapping(
         else:
             self.modification_mapping = copy.deepcopy(modification_mapping)
 
-        self._mods_as_lists()
-        self._reverse_mod_mapping()
+        self._str_mods_to_lists()
```

Comment:
Is there a reason why this attribute is mutated in place, while the others are reassigned?
(see e.g. line 269)

Reply:
I guess for readability? Complex Inplace operations are hard to read and track

Reply:
good catch @lucas-diedrich , I was wondering too .. it's as @jalew188 suspected indeed the limited complexity that made me keep `str_mods_to_lists` as in-place .. but more based on a gut feeling rather than on a strict reasoning .. I could change it for consistency if you like?

Reply:
I see, thank you for the explanation 👍 So I guess the alternative would then be something like 
```python
self.modification_mapping = _str_mods_to_lists(self.modification_mapping)
```

With
```python
def _str_mods_to_lists(self, modification_mapping: dict) -> dict[str, list[str]]:
     """Convert all single strings to lists containing one item in self.modification_mapping."""
     for mod, val in list(modification_mapping.items()):
         if isinstance(val, str):
             modification_mapping[mod] = [val]
     return modification_mapping
```

I don't have a strong opinion on this, and the current way looks great, so no need to change it!